webrender/res/cs_border_segment.glsl

```diff
@@ -227,6 +225,17 @@ void main(void) {
 
         vPos = vClipParams1.xy + radius * (2.0 * aPosition.xy - 1.0);
         vPos = clamp(vPos, vec2(0.0), aRect.zw);
+    } else if (clip_mode == CLIP_DASH_CORNER) {
+        vec2 center = (aClipParams1.xy + aClipParams2.xy) * 0.5;
+        // This is a gross approximation which works out because dashes don't have
+        // a strong curvature and we will overshoot by inflating the geometry by
+        // this amount on each side (sqrt(2) * length(dash) would be enough and we
+        // compute 2 * approx_length(dash)).
+        float dash_length = length(aClipParams1.xy - aClipParams2.xy);
```

Comment:
is it possible to process width and height separately here (and get a tighter bound therefore)?

Reply:
I may be missing something but since we don't know the orientation of the dash, it's hard to consider the dash width and length separately. So we have to assume the longest side in any direction and potentially the worst orientation (45 degrees).

We could compute the orientation from `aClipParams1.xy`  and `aClipParams2.xy` which are at the start and end of the dash but that would add a lot of math. To get a really tight rect we'd need to consider the curvature of the dash in the computation which would make that a ton more complicated for a rather small amount of overdraw.

That's why I settled for a very conservative rect. Without extra information we could technically reduce the rect because, again, assuming dashes are rectangles with no curvature (which is not quite true), a sufficiently conservative bounding square would have each side equal to `sqrt(2) * max(width, length)` (sqrt(2) coming from the worst-case angle of 45 degrees) and this PR uses a factor of 2 instead of sqrt(2). The factor of two is implicit (it's just that we inflate the rect of that quantity on each side).

Because this formula doesn't account for the curvature we'd need some extra headroom to make sure we don't clip dashes with strong curvature but quantifying how much we need is not worth the complexity in my opinion. It's most likely less than the ~30% from this PR though. So if we are willing to accept a magic constant factor that has no mathematical ground, we should be safe multiplying by, say, 0.8 and leave only about 10% on each side for the curvature. I almost put that on the PR but I suspected an arbitrary factor would generate some debate.

I'm not sure real-world pages will benefit much from tighter bounding rects anyway, so I think we should stick to something simple and maybe come back and implement something more complicated if/when we have evidence that it can have an impact.

Reply:
So right now there isn't really a magic factor (other than since we are creating a square with edges at `±max(width, length)` from the center we get `2*max(width, length)` sides. Do you mean adding a factor like the 0.8 I previously mentioned or do you mean I should reword the existing comment?